core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -1189,7 +1189,7 @@ protected function getNewPosition($mode, $pid=null, $insertInto=false)
 
 					if ($limit > 0)
 					{
-						$objInsertAfter = $this->Database->prepare("SELECT id FROM " . $this->strTable . " WHERE pid=? ORDER BY sorting, id")
+						$objInsertAfter = $this->Database->prepare("SELECT id FROM " . $this->strTable . " WHERE " . ($pid == 0 ? '(pid=? OR pid IS NULL)' : 'pid=?') . " ORDER BY sorting, id")
```

Comment:
```suggestion
						$objInsertAfter = $this->Database->prepare("SELECT id FROM " . $this->strTable . " WHERE COALESCE(pid, 0)=? ORDER BY sorting, id")
```
maybe this could be a simpler version? (untested)

Reply:
feels less readable and not sure about performance. If there is no technical advantage I don't think we should use that.

Reply:
We could also use `IFNULL()` if `COALESCE()` is less readable.

```suggestion
						$objInsertAfter = $this->Database->prepare("SELECT id FROM " . $this->strTable . " WHERE IFNULL(pid, 0)=? ORDER BY sorting, id")
```

But `IFNULL(pid, 0)=?` is way more readable then `" . ($pid == 0 ? '(pid=? OR pid IS NULL)' : 'pid=?') . "` don’t you think?

Reply:
do you think that's more/equally performant as a simple OR statement?

Reply:
Yes. (But I also think that the performance for that part is not relevant here)
But I just saw that we’d need to change `->execute($pid)` then to `->execute((int) $pid)` too, so maybe that better then.

How about at least removing the scary `== 0` part?

```suggestion
						$objInsertAfter = $this->Database->prepare("SELECT id FROM " . $this->strTable . " WHERE " . ($pid ? 'pid=?' : '(pid=? OR pid IS NULL)') . " ORDER BY sorting, id")
```

Reply:
> How about at least removing the scary == 0 part?

`$pid` in this case refers to a string `'0'` which would not equal to boolean false, so that won't work.